### ISSUE TITLE
How can I use this code for training and testing?

### ISSUE THREAD

**Issue by daigang896** [tjiiv-cprg/SPFCN-ParkingSlotDetection#8]
Hello. How can I use this code for training and testing?

**Comment by yjyjy131** [comment]
I modified the code a bit and requested a pull-request make it easier to train and test the author's SPFCN (not light version), but I'm not sure if this code will be accepted by the author. 
If not, please see this one. [https://github.com/yjyjy131/SPFCN-ParkingSlotDetection](https://github.com/yjyjy131/SPFCN-ParkingSlotDetection). 
And please share issues if you have some !

**Comment by wangzh63** [comment]
> I modified the code a bit and requested a pull-request make it easier to train and test the author's SPFCN (not light version), but I'm not sure if this code will be accepted by the author. If not, please see this one. https://github.com/yjyjy131/SPFCN-ParkingSlotDetection. And please share issues if you have some !

Process is killed during the training, have you met this problem before? thank you in advance

**Comment by yjyjy131** [comment]
> > I modified the code a bit and requested a pull-request make it easier to train and test the author's SPFCN (not light version), but I'm not sure if this code will be accepted by the author. If not, please see this one. https://github.com/yjyjy131/SPFCN-ParkingSlotDetection. And please share issues if you have some !
> 
> Process is killed during the training, have you met this problem before? thank you in advance

Yes, I have. The process was killed when I tried it with a computer running out of GPU memory. Can you tell me exactly what error you're getting?

If it is an out of memory problem, there seem to be two ways.

One trick (although this is not the right way) was to delete validation process in the training the code and reduce the training batch size or num_workers size, so that it could be trained even on a computer with insufficient memory.

There is currently no SPFCN-light version of the training code, so if you do not have enough GPU, it would be better to write and use the light version of the training code.

**Comment by wangzh63** [comment]
I think the error shows up during the process of creating the training dataset, It seems like I do not have enough memory to load it.

**Comment by wangzh63** [comment]
> > > I modified the code a bit and requested a pull-request make it easier to train and test the author's SPFCN (not light version), but I'm not sure if this code will be accepted by the author. If not, please see this one. https://github.com/yjyjy131/SPFCN-ParkingSlotDetection. And please share issues if you have some !
> > 
> > 
> > Process is killed during the training, have you met this problem before? thank you in advance
> 
> Yes, I have. The process was killed when I tried it with a computer running out of GPU memory. Can you tell me exactly what error you're getting?
> 
> If it is an out of memory problem, there seem to be two ways.
> 
> One trick (although this is not the right way) was to delete validation process in the training the code and reduce the training batch size or num_workers size, so that it could be trained even on a computer with insufficient memory.
> 
> There is currently no SPFCN-light version of the training code, so if you do not have enough GPU, it would be better to write and use the light version of the training code.

I think the error shows up during the process of creating the training dataset, It seems like I do not have enough memory to load it.

**Comment by wslyw** [comment]
> > > I modified the code a bit and requested a pull-request make it easier to train and test the author's SPFCN (not light version), but I'm not sure if this code will be accepted by the author. If not, please see this one. https://github.com/yjyjy131/SPFCN-ParkingSlotDetection. And please share issues if you have some !
> > 
> > 
> > Process is killed during the training, have you met this problem before? thank you in advance
> 
> Yes, I have. The process was killed when I tried it with a computer running out of GPU memory. Can you tell me exactly what error you're getting?
> 
> If it is an out of memory problem, there seem to be two ways.
> 
> One trick (although this is not the right way) was to delete validation process in the training the code and reduce the training batch size or num_workers size, so that it could be trained even on a computer with insufficient memory.
> 
> There is currently no SPFCN-light version of the training code, so if you do not have enough GPU, it would be better to write and use the light version of the training code.

Hi, do you use Ps2.0 for your dataset? 
``

**Comment by wslyw** [comment]
> I modified the code a bit and requested a pull-request make it easier to train and test the author's SPFCN (not light version), but I'm not sure if this code will be accepted by the author. If not, please see this one. https://github.com/yjyjy131/SPFCN-ParkingSlotDetection. And please share issues if you have some !

Hi, is the dataset you are using Ps2.0 dataset?

**Comment by MihaiBlagu** [comment]
Hi @yjyjy131, did you encounter key errors when loading the state_dict of the trained model? I get a bunch of missing keys and some unexpected ones too.

I am loading the merge_bn_epoch150_loss.pkl (I trained for 150 epochs).
